src/packet/pacer.rs

```diff
@@ -280,85 +281,71 @@ impl Pacer for LeakyBucketPacer {
     }
 
     fn poll_timeout(&self) -> Option<Instant> {
-        if self.need_immediate_timeout {
-            return self.last_send_time;
-        }
-
         let next_handle_time = self.last_handle_time.map(|lh| lh + PACING);
 
-        match (next_handle_time, self.next_send_time) {
-            (Some(nh), Some(ns)) => Some(nh.min(ns)),
-            (None, Some(ns)) => Some(ns),
+        match (next_handle_time, self.next_poll_time) {
+            (Some(nh), Some(np)) => Some(nh.min(np)),
+            (None, Some(np)) => Some(np),
             (Some(nh), None) => Some(nh),
             (None, None) => None,
         }
     }
 
-    fn handle_timeout(&mut self, now: Instant, iter: impl Iterator<Item = QueueState>) {
-        self.need_immediate_timeout = false;
+    fn handle_timeout(
+        &mut self,
+        now: Instant,
+        iter: impl Iterator<Item = QueueState>,
+    ) -> Option<PaddingRequest> {
+        // This is called periodically and whenever packet is queued.
         self.queue_states.clear();
         self.queue_states.extend(iter);
 
-        // This is called periodically and whenever packet is queued.
         let elapsed = self.update_handle_time_and_get_elapsed(now);
 
         self.clear_debt(elapsed);
         self.maybe_update_adjusted_bitrate(now);
-        let should_recalcualte =
-            matches!(self.next_poll_outcome, Some(PollOutcome::Nothing) | None);
 
-        if !should_recalcualte {
-            return;
+        if let Some(request) = self.maybe_create_padding_request() {
+            // We need refreshed queue state to generate an action to poll the generate padding.
+            self.request_immediate_timeout();
```

Comment:
Why do we need another timeout? We just got one!

Reply:
Because at this point the queue state is entirely empty, otherwise we wouldn't have requested padding. Now, to actually send the padding we need a new handle_timeout with a refreshed queue state so we can emit a poll for the relevant queue

Reply:
I was thinking we could just set `next_poll_queue` here, but I decided against it...

Reply:
Since poll_output follows anyway, can't you just say that you want a poll on the `mid` you just generated padding for?

Reply:
Why do we need a refreshed queue state?

Reply:
My concern was around starving out media but maybe that's not gonna be a problem 🤔

Reply:
Changed it